src/primitives/checksum.rs

```diff
@@ -327,4 +495,42 @@ mod tests {
         assert_eq!(packed.unpack(2), 2);
         assert_eq!(packed.unpack(3), 1);
     }
+
+    #[test]
+    #[cfg(feature = "alloc")]
+    fn bech32() {
+        // In codes that Pieter specifies typically the generator polynomial is
+        // only given indirectly, in the reference code which encodes it in a
+        // packed form (shifted multiple times).
+        //
+        // For example in the BIP173 Python reference code you will see an array
+        // called `generator` whose first entry is 0x3b6a57b2. This first entry
+        // is the generator polynomial in packed form.
+        //
+        // To get the expanded polynomial form you can use `u128::unpack` like so:
+        let unpacked_poly = (0..6)
+            .rev() // Note .rev() to convert from BE integer literal to LE polynomial!
+            .map(|i| 0x3b6a57b2u128.unpack(i))
+            .map(|u| Fe32::try_from(u).unwrap())
+            .collect::<Vec<_>>();
+        assert_eq!(unpacked_poly, [Fe32::A, Fe32::K, Fe32::_5, Fe32::_4, Fe32::A, Fe32::J],);
+        // To get a version of the above with bech32 chars instead of Fe32s, which
+        // can be a bit hard to print, just stick a `.map(Fe32::to_char)` into the
+        // above iterator chain.
+
+        // Ok, exposition over. The actual unit test follows.
+
+        // Run with -- --nocapture to see the output of this. This unit test
+        // does not check the exact output because it is not deterministic,
+        // and cannot check the code semantics because Rust does not have
+        // any sort of `eval`, but you can manually check the output works.
+        let _s = PrintImpl::<Fe1024>::new(
+            "Bech32",
+            &[Fe32::A, Fe32::K, Fe32::_5, Fe32::_4, Fe32::A, Fe32::J],
```

Comment:
Can we pass `&unpacked_poly` here directly, to avoid duplication/confusion?

Reply:
I'll try but IIRC there was some type confusion that prevented me from doing so

Reply:
Oh, I see, you mean to take the `unpacked_poly` computation from the unit test and put it in the doctest, so that the only magic constants that appear clearly came from the BIP. Sure.